Library/Homebrew/cmd/test-bot.rb

```diff
@@ -549,15 +549,13 @@ def test(*args)
     end
 
     def check_results
-      status = :passed
-      steps.each do |step|
+      steps.all? do |step|
         case step.status
-        when :passed  then next
+        when :passed  then true
         when :running then raise
-        when :failed  then status = :failed
+        when :failed  then false
         end
       end
-      status == :passed
     end
```

Comment:
Does this produce the same results but just makes the function slightly more efficient?